### ISSUE TITLE
Something wrong with Jupyter token or password in the latest docker image?

### ISSUE THREAD

**Issue by seisman** [mspass-team/mspass#185]
I followed the instructions at https://wangyinz.github.io/mspass/getting_started/run_mspass_with_docker.html, and ran:
```
docker pull wangyinz/mspass
docker run -p 8888:8888 --mount src=`pwd`,target=/home,type=bind wangyinz/mspass
```
Here is the output:
![image](https://user-images.githubusercontent.com/3974108/115896807-cde6c700-a429-11eb-80c0-9a21e43fcc98.png)

When I click the `http://127.0.0.1:8888/?token=xxxxxx` link, it should open the jupyter notebook. It works before, but today, when I click the link, the web browser shows:
![image](https://user-images.githubusercontent.com/3974108/115897122-1900da00-a42a-11eb-9dc2-0fb21ebac9b1.png)

I also tried to paste the token into the box, but it still doesn't open the jupyter notebook.

Is this caused by incorrect jupyter settings in the docker image? Or am I doing something wrong?

**Comment by wangyinz** [comment]
When you manually put in the token, did it say "Invalid credentials"? That should show up if the token was incorrect. 

Also, I wonder if you have some other Jupyter notebook running at the same time. Have you tried binding the container's 8888 port to a different port on the host? e.g., you can bind it to port 9999 with

```
docker run -p 9999:8888 wangyinz/mspass
```

You can also use `docker ps` to check if there is any other containers runnin.

**Comment by seisman** [comment]
> I wonder if you have some other Jupyter notebook running at the same time.

Yes, that's the reason! After closing other notebooks, it works now. 

Thanks!

**Comment by seisman** [comment]
Perhaps it's better to recommend other another port (e.g., 9999) since users who run their local Jupyter always use the 8888 port by default.

**Comment by wangyinz** [comment]
Well, `8888` is the default port from Jupyter. It will be more confusing to have a different default in MsPASS, and the document does have instruction on how to bind to a different port. So, I believe it is better to keep it as is.

**Comment by seisman** [comment]
I'm reopening the issue, because the documentation is still a little unclear.

In this section (http://www.mspass.org/getting_started/run_mspass_with_docker.html), it says:

> If there are collisions with 8888 port on the host, change the first number (e.g. to use port 9999 on the host use -p 9999:8888.)
> Now, you can use the link printed on the output that starts with http://127.0.0.1/ to access the notebook and run codes within the container. 

The problem is, when using `-p 9999:8888`, Jupyter notebook in the docker image uses the 8888 port and prints a URL like `http://127.0.0.1:8888/?token=XXX`, but the host uses the 9999 port. Clicking the link `http://127.0.0.1:8888/?token=XXX` can't open the Jupyter notebook, and users have to change 8888 to 9999 in the link.

It would be better to rephrase some sentences to make it clear how to use a different port correctly.

**Comment by wangyinz** [comment]
Good point. We should revise that.

**Comment by wangyinz** [comment]
Closing this as completed.

The current Docker guide explicitly explains that `-p 9999:8888` changes only the host port and instructs users to replace `8888` with `9999` in the JupyterLab URL. The command-line desktop guide gives the same `http://127.0.0.1:9999` example, resolving the ambiguity reported here.
